src/permissions/permissions.ts

```diff
@@ -0,0 +1,92 @@
+import { PackedItem } from '@/item/packedItem.js';
+import { CompleteMember, MemberType } from '@/member/member.js';
+
+// TODO: reuse action op?
+export enum Operation {
+  ReadItem = 'read-item',
+  MoveItem = 'move-item',
+  CopyItem = 'copy-item',
+  HideItem = 'hide-item',
+  DownloadItem = 'download-item',
+}
+
+// don't take into account hidden: suppose backend will prevent read from beginning
+// pseudonymized users cannot create items and can only view a specific item
+const SIGNED_IN_OPERATIONS_PERMISSIONS = {
+  [Operation.ReadItem]: {
+    admin: true,
+    write: true,
+    read: true,
+    pseudonymized: true,
+    public: true,
+  },
+  [Operation.CopyItem]: {
+    admin: true,
+    write: true,
+    read: true,
+    pseudonymized: false,
+    public: true,
+  },
```

Comment:
This PR is 🔥, I love it !
What about using namespaces so we reduce the amount of operations ? We can also have an enumeration of roles so we can type check it !

```suggestion
export enum Operation {
  Read = 'read',
  Move = 'move',
  Copy = 'copy',
  Hide = 'hide',
  Download = 'download',
}

export enum Namespace {
  Item = 'item',
  Membership = 'Membership',
  Analytics = 'Analytics',
  Profile = 'Profile'
}

export enum Role {
  Admin = 'admin',
  Write = 'write',
  Read = 'read',
  Pseudonymized = 'pseudonymized',
  Public = 'public'
}

// don't take into account hidden: suppose backend will prevent read from beginning
// pseudonymized users cannot create items and can only view a specific item
const SIGNED_IN_OPERATIONS_PERMISSIONS: { [key in Namespace]?: { [key in Operation]?: { [key in Role]?: boolean } } } = {
  [Namespace.Item]: {
    [Operation.Read]: {
      [Role.Admin]: true,
      [Role.Write]: true,
      [Role.Read]: true,
      [Role.Pseudonymized]: true,
      [Role.Public]: true,
    },
    [Operation.Copy]: {
      [Role.Admin]: true,
      [Role.Write]: true,
      [Role.Read]: true,
      [Role.Pseudonymized]: false,
      [Role.Public]: true,
    },
```